### PR TITLE
Use sigchain v2 for track and untrack links

### PR DESCRIPTION
**Pull request by joshblum**
Use [sigchain v2](https://keybase.io/docs/teams/sigchain_v2) for track and untrack links in user sig chains!

### CODE REVIEW

File: go/libkb/user.go [hunk 1]
```diff
@@ -712,23 +719,6 @@ func (u *User) SigningKeyPub() (GenericKey, error) {
 	return pubKey, nil
 }
 
-func (u *User) TrackStatementJSON(them *User, outcome *IdentifyOutcome) (string, error) {
```
Comment (joshblum) [inline go/libkb/user.go]: removed since unused

**Comment by joshblum** [comment]
Diff is kinda huge since all relevant tests run against v1 and v2...

**Comment by maxtaco** [comment]
Can you not change the indentation on the tests? You can do so with a helper function. It will make the diff smaller and clear you of git blames for al those changes.

File: go/engine/track_token.go [hunk 2]
```diff
@@ -219,31 +226,60 @@ func (e *TrackToken) storeRemoteTrack(ctx *Context, pubKID keybase1.KID) (err er
...
+	var sig string
+	var sigid keybase1.SigID
+	sigVersion := libkb.SigVersion(e.arg.Options.SigVersion)
+	switch sigVersion {
+	case libkb.KeybaseSignatureV1:
+		sig, sigid, err = signingKey.SignToString(e.trackStatementBytes)
+		linkID = libkb.ComputeLinkID(e.trackStatementBytes)
+	case libkb.KeybaseSignatureV2:
+		prevSeqno := me.GetSigChainLastKnownSeqno()
+		prevLinkID := me.GetSigChainLastKnownID()
```
[the remark this answers is not in the document]
Comment (joshblum) [reply go/engine/track_token.go]: I left it as is since those fields are used for the teams signatures differently

**Comment by jzila** [comment]
Ah, looks like Jenkins never ran for this.

**Comment by jzila** [comment]
I think this was when Jenkins was broken yesterday, during the restart.

**Pull request merged**